jl4-core/src/L4/TypeCheck.hs

```diff
@@ -143,11 +142,14 @@ doCheckProgramWithDependencies checkState moduleUri program =
 --
 combineEnvironmentEntityInfo :: Environment -> EntityInfo -> [(Name, CheckEntity)]
 combineEnvironmentEntityInfo env ei =
-  let
-    uniques :: Set Unique
-    uniques = Set.fromList (concat (snd <$> Map.toList env))
-  in
-    snd <$> filter (flip Set.member uniques . fst) (Map.toList ei)
+    foldMap (uncurry lookupUnqs) $ Map.toList env
+    where
+      lookupUnqs rn unqs = foldMap (maybeToList . fmap (first (replaceRawName rn)) . (ei Map.!?)) unqs
```

Comment:
```suggestion
      lookupUniques rn uniques = foldMap (maybeToList . fmap (first (replaceRawName rn)) . (ei Map.!?)) uniques
```

Reply:
I don't think the abbrevation is worth it :)

Reply:
Also, might be worth splitting over multiple lines.

Reply:
Type signature might be helpful as well.
This line of code feels overcomplicated to me.